src/dali_executor/dali_pipeline.cc

```diff
@@ -26,14 +26,27 @@
 
 namespace triton { namespace backend { namespace dali {
 
+std::vector<int64_t>
+dali_to_list_shape(const TensorListShape<>& shape)
+{
+  auto dim = shape.ndim;
+  auto batch_size = shape.num_samples();
+  std::vector<int64_t> result(dim * batch_size);
+  for (int64_t s = 0; s < batch_size; ++s) {
+    for (int64_t d = 0; d < dim; ++d) {
+      result[s * dim + d] = shape[s][d];
+    }
+  }
+  return result;
```

Comment:
```suggestion
  return shape.shapes;
```
Note that this doesn't work for 0D tensor lists.
Also - do you really need to make this copy?

Reply:
Nope the copy was not needed. Removed this function